### PR TITLE
Emit metrics on job starts and ends, with status as name and job id a…

### PR DESCRIPTION
**Pull request by jonmv**
…s dim.

@mpolden please review. 
Any comments on metric names, @frodelu @yngveaasheim ?

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import com.yahoo.config.provision.SystemName;
+import com.yahoo.jdisc.Metric;
+import com.yahoo.vespa.hosted.controller.api.integration.deployment.JobId;
+
+import java.util.Map;
+
+/**
+ * Records metrics related to deployment jobs.
+ *
+ * @author jommv
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]: Typo.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package com.yahoo.vespa.hosted.controller.deployment;
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]: Missing copyright header:

`// Copyright 2020 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.`

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    static String valueOf(RunStatus status) {
+        switch (status) {
+            case outOfCapacity: return outOfCapacity;
+            case deploymentFailed: return deploymentFailure;
+            case installationFailed: return convergenceFailure;
+            case testFailure: return testFailure;
+            case error: return error;
+            case aborted: return abort;
+            case success: return success;
+            case running: throw new IllegalArgumentException("Not supposed to get this value");
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]: Include the unexpected value in the message (or remove this case and let `default` handle it).

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]: Nah, changed my name last week >)<

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]:
```suggestion
 * @author jonmv
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]:
```suggestion
// Copyright 2020 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.vespa.hosted.controller.deployment;
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java (hunk 3 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]: I have no idea why I wrote it this way o_O

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobMetrics.java]:
```suggestion
```

**Review by mpolden — APPROVED** [review]

**Pull request merged**